sdk/schemaregistry/azure-schemaregistry-avroencoder/azure/schemaregistry/encoder/avroencoder/_schema_registry_avro_encoder.py

```diff
@@ -271,38 +324,78 @@ def decode(
             try:
                 content = message["content"]
                 content_type = message["content_type"]
-            except (KeyError, TypeError):
-                SchemaDecodeError(
-                    f"""The content model {str(message)} is not a subtype of the MessageType protocol or type
-                        MessageContent.  If using an Azure SDK model class, please check the README.md
+            except (KeyError, TypeError) as exc:
+                raise TypeError(
+                    f"""The content model {str(message)} must be a subtype of the MessageType protocol or type
+                        MessageContent. If using an Azure SDK model class, please check the README.md
                         for the full list of supported Azure SDK models and their corresponding versions."""
-                ).raise_with_traceback()
+                ) from exc
 
-        # include in first preview for back compatibility
-        content, content_type = self._convert_preamble_format(content, content_type)
-
-        schema_id = content_type.split("+")[1]
+        try:
+            schema_id = content_type.split("+")[1]
+        except AttributeError:
+            raise InvalidContentError(
+                "Content type was not in the expected format of MIME type + schema ID."
+            )
```

Comment:
I see this check is newly added, I personally like it because it tells me the right format of a schema, however it makes me wonder whether we should apply stricter client validation on the content_type: e.g. "a+b+c" is invalid

maybe we could simply handle it by also checking the length of array after splitting should be 2?

Reply:
yep, that sounds fair to me